src/test/groovy/uk/gov/hmcts/dm/it/ReadDocumentIT.groovy

```diff
@@ -239,35 +239,102 @@ class ReadDocumentIT extends BaseIT {
+            .when()
+            .get(documentUrl)
 
     }
 
+    @Test
+    void "R17 As authenticated user with a specific role I can access a document if its CLASSIFICATION is public and matches role"() {
+
+        createUser(CITIZEN_2, 'caseworker')
 
+        def documentUrl = createDocumentAndGetUrlAs CITIZEN, ATTACHMENT_1, 'PUBLIC', ['citizen', 'caseworker']
+
+        givenRequest(CITIZEN_2)
+            .expect()
+            .statusCode(200)
+            .when()
+            .get(documentUrl)
+    }
 
     @Test
-    void "R14 As authenticated user with a specific role I can access a document if its PUBLIC is restricted and roles match"() {
+    void "R18 As authenticated user with no role I cannot access a document if its CLASSIFICATION is public with no role"() {
+
+        createUser(CITIZEN_2, null)
+
+        def documentUrl = createDocumentAndGetUrlAs CITIZEN, ATTACHMENT_1, 'PUBLIC', [null]
+
+        givenRequest(CITIZEN_2)
+            .expect()
+            .statusCode(403)
+            .when()
+            .get(documentUrl)
+    }
+
+    @Test
+    void "R19 As authenticated user with some role I can access a document if its CLASSIFICATION is public and roles does not match"() {
 
         createUser(CITIZEN_2, 'caseworker')
 
-        def documentUrl = createDocumentAndGetUrlAs CITIZEN, ATTACHMENT_1, 'PUBLIC', ['caseworker']
+        def documentUrl = createDocumentAndGetUrlAs CITIZEN, ATTACHMENT_1, 'PUBLIC', [null]
+
+        givenRequest(CITIZEN_2)
+            .expect()
+            .statusCode(403)
+            .when()
+            .get(documentUrl)
+
+    }
+
+    @Test
+    void "R20 As authenticated user with no role (Tests by default sets role as citizen) I can access a document if its CLASSIFICATION is public and roles is citizen"() {
+
+        createUser(CITIZEN_2, null)
+
+        def documentUrl = createDocumentAndGetUrlAs CITIZEN, ATTACHMENT_1, 'PUBLIC', ['citizen']
+
+        givenRequest(CITIZEN_2)
+            .expect()
+            .statusCode(200)
+            .when()
+            .get(documentUrl)
+    }
+
+    @Test
+    void "R21 As an owner I can access a document even if its CLASSIFICATION is private with no roles"() {
+
+        createUser(CITIZEN_2, null)
+
+        def documentUrl = createDocumentAndGetUrlAs CITIZEN_2, ATTACHMENT_1, 'PRIVATE', [null]
+
+        givenRequest(CITIZEN_2)
+            .expect()
+            .statusCode(200)
+            .when()
+            .get(documentUrl)
+    }
+
+    @Test
+    void "R22 As authenticated user with a specific role I can't access a document if its CLASSIFICATION is restricted and roles don't match"() {
+
+        createUser(CITIZEN_2, 'citizen')
+
+        def documentUrl = createDocumentAndGetUrlAs CITIZEN, ATTACHMENT_1, 'RESTRICTED', ['caseworker']
 
         givenRequest(CITIZEN_2)
                 .expect()
-                .statusCode(200)
+                .statusCode(403)
                 .when()
                 .get(documentUrl)
-
     }
```

Comment:
If I am a user with no roles
I create a document with casework role + classification is private
then should I be able to get this file 

I need a test to prove this works please

Reply:
Added the requested test in the integration tests